File: TV.rb
class TV

  def initialize()
    @estado = false
    @canal = 1
    @volume = 0
  end

  def ligar()
    @estado = true
  end

  def desligar()
    @estado = false
  end

  def alterar_canal(novo_canal)
    validar_canal(novo_canal)
    @canal = novo_canal
  end

  def aumentar_volume()
    @volume += 1
    validar_volume()
  end

  def diminuir_volume()
    @volume -= 1
    validar_volume()
  end

  def consultar_estado()
    @estado
  end

  def consultar_canal()
    @canal
  end

  def consultar_volume()
    @volume
  end

   PRIMEIRO_CANAL = 1
   ULTIMO_CANAL = 60

private

  def validar_canal(novo_canal)
    raise if novo_canal < PRIMEIRO_CANAL || novo_canal > ULTIMO_CANAL
  end

  def validar_volume()
    raise if @volume < 0 || @volume > 30
  end
end

File: TV.java

public class TV {
	
	private int canal;
	private int volume;
	private boolean ligada;
	public static final int PRIMEIRO_CANAL = 1;
	public static final int ULTIMO_CANAL = 60;
	public static final int LIMITE_INFERIOR_VOLUME = 0;
	public static final int LIMITE_SUPERIOR_VOLUME = 30;

	public TV(){
		this.canal = 1;
		this.volume = 0;
		this.ligada = false;
	}

	public int canal() {
		return this.canal;
	}

	public int volume() {
		return this.volume;
	}
	
	public boolean ligada(){
		return this.ligada;
	}
	
	public void ligar(){
		this.ligada = true;
	}
	
	public void desligar(){
		this.ligada = false;
	}

	public void canalAcima() {
		if (this.canal < ULTIMO_CANAL)
			this.canal += 1;		
	}
	
	public void canalAbaixo() {
		if (this.canal > PRIMEIRO_CANAL)
			this.canal -= 1;		
	}

	public void aumentarVolume() {
		if (this.volume < LIMITE_SUPERIOR_VOLUME)
			this.volume += 1;		
	}
	
	public void diminuirVolume() {
		if (this.volume > LIMITE_INFERIOR_VOLUME)
			this.volume -= 1;		
	}
}
